python/oneseismic/client/client.py

```diff
@@ -141,20 +148,12 @@ def token(self):
             except Exception as e:
                 raise ConfigError('Bad config') from e
 
-            tokpath = tokenpath(self.cache_dir)
-            cache = msal.SerializableTokenCache()
-
-            with open(tokpath) as tokenfile:
-                cache.deserialize(tokenfile.read())
-            atexit.register(
-                lambda: open(tokpath, "w").write(cache.serialize())
-            )
-
-            self.app = msal.PublicClientApplication(
-                config['client_id'],
-                authority=config['auth_server'],
-                token_cache=cache,
-            )
+            with tokencache(self.cache_dir) as token_cache:
```

Comment:
At the end of this context the token has not yet been updated. This will result in the same token being written back to disk as was loaded from disk, not the updated token.